Undefined reference to curl@CURL_OPENSSL

# Please follow the general troubleshooting steps first:

- [X] Ran `brew update` and retried your prior step?
- [X] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [X] Confirmed this is a problem with specific formulae and not brew? If it's a general brew problem please file this issue at https://github.com/supernemo-dbd/brew/issues/new

_You can erase any parts of this template not applicable to your Issue._

### Bug reports:

After a successful ```brew update``` (so everything got installed well), I got link problems when building bayeux from source. Typical error messages are

```
[661/665] Linking CXX executable BuildProducts/bin/bxg4_production-3.0.0
/home/garrido/Workdir/NEMO/supernemo/snware/brew/lib/libxerces-c-3.1.so : undefined reference to « curl_easy_getinfo@CURL_OPENSSL_4 »
/home/garrido/Workdir/NEMO/supernemo/snware/brew/lib/libxerces-c-3.1.so : undefined reference to « curl_multi_perform@CURL_OPENSSL_4 »
/home/garrido/Workdir/NEMO/supernemo/snware/brew/lib/libxerces-c-3.1.so : undefined reference to « curl_multi_cleanup@CURL_OPENSSL_4 »
```

I also tried from a fresh installation of ```brew/cadfael``` from scratch with ```build-from-source``` flag to avoid the use of bottles and finally got the same issue. 

Here is the ```brew config```

```
HOMEBREW_VERSION: >1.2.0 (no git repository)
ORIGIN: https://github.com/SuperNEMO-DBD/brew.git
HEAD: 2bdfd11fd2663f58e6f4a63986344dd777b7c70f
Last commit: il y a 6 jours
Core tap ORIGIN: https://github.com/Linuxbrew/homebrew-core
Core tap HEAD: d48874b4184ca4eb8ef803d1cb16323292c75f5d
Core tap last commit: il y a 10 heures
HOMEBREW_PREFIX: /home/garrido/Workdir/NEMO/supernemo/snware/brew
HOMEBREW_REPOSITORY: /home/garrido/Workdir/NEMO/supernemo/snware/brew
HOMEBREW_CELLAR: /home/garrido/Workdir/NEMO/supernemo/snware/brew/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://linuxbrew.bintray.com
CPU: quad-core 64-bit ivybridge
Homebrew Ruby: 2.4.2 => /usr/bin/ruby
Clang: N/A
Git: 2.14.1 => /usr/bin/git
Perl: /usr/bin/perl
Python: /home/garrido/Workdir/NEMO/supernemo/snware/brew/opt/python/libexec/bin/python => /home/garrido/Workdir/NEMO/supernemo/snware/brew/Cellar/python/2.7.13_1/bin/python2.7
Ruby: /usr/bin/ruby
Java: 1.7.0_151
Kernel: Linux 4.12.13-1-ARCH x86_64 GNU/Linux
OS: Arch Linux (n/a)
Host glibc: 2.26
/usr/bin/gcc: 7.2.0
glibc: N/A
gcc: N/A
xorg: N/A
```

--- a/Formula/bayeux.rb
+++ b/Formula/bayeux.rb
@@ -3,7 +3,7 @@ class Bayeux < Formula
   homepage "https://github.com/supernemo-dbd/bayeux"
   url "https://files.warwick.ac.uk/supernemo/files/Cadfael/distfiles/Bayeux-3.0.0.tar.bz2"
   sha256 "f0f01465ad20e51a05ca889cfdc52c12d0a2cf16d70dd6f4e04551c652ce967e"
-  revision 1
+  revision 2
 
   option "with-devtools", "Build debug tools for Bayeux developers"
 

--- a/Formula/falaise.rb
+++ b/Formula/falaise.rb
@@ -3,7 +3,7 @@ class Falaise < Formula
   homepage "https://supernemo-dbd.github.io"
   url "https://files.warwick.ac.uk/supernemo/files/Cadfael/distfiles/Falaise-3.0.0.tar.bz2"
   sha256 "2cba670ce626887270af5f6e98106c6f07c5f3214fb281af5ef7894442d544d7"
-  revision 1
+  revision 2
 
   head "https://github.com/SuperNEMO-DBD/Falaise.git", :branch => "develop"
 

--- a/Formula/geant4.rb
+++ b/Formula/geant4.rb
@@ -1,7 +1,7 @@
 class Geant4 < Formula
   desc "C++ toolkit for simulating the passage of particles through matter"
   homepage "http://geant4.cern.ch"
-  revision 4
+  revision 5
 
   stable do
     url "http://geant4.cern.ch/support/source/geant4.9.6.p04.tar.gz"

--- a/Formula/xerces-c.rb
+++ b/Formula/xerces-c.rb
@@ -3,6 +3,9 @@ class XercesC < Formula
   homepage "https://xerces.apache.org/xerces-c/"
   url "https://www.apache.org/dyn/closer.cgi?path=xerces/c/3/sources/xerces-c-3.1.4.tar.gz"
   sha256 "c98eedac4cf8a73b09366ad349cb3ef30640e7a3089d360d40a3dde93f66ecf6"
+  revision 1
+
+  depends_on "curl" unless OS.mac?
 
   option :cxx11
 

